doc/model_concepts.rst

```diff
@@ -0,0 +1,141 @@
+                                                      associated with a unique cell, identified by the member `gid`,
+                                                      and identifying an item within a cell-local collection by the
+                                                      member `index`.
+    ========================  ======================  ===========================================================
+
+
+Each cell has a global identifier :cpp:type:`gid` associated to it. The :cpp:type:`gid` is used by the :ref:`recipe <modelrecipe>`
+to build the global network. Cells may interact with other cells via :ref:`connections <modelconnections>` or
+:ref:`gap junctions <modelgapjunctions>`. These interactions happen at specific positions on the cell which can
+be one of three types:
+
+1. **Source**: A spike source associated to a *connection*. It detects spikes; if part of a connection, it sends the spikes to **targets**.
+2. **Target**: A synapse associated to a *connection*. It has an associated synaptic mechanism that can respond to spikes from **sources**.
+3. **Gap Junction Site**: A site associated to a *gap junction*. It represents one half of the gap junction.
+
+Each of these positions is located on a specific point on a cell. It has an :cpp:type:`index` relative to all other positions of
+the same type on that cell. That cell, as previously mentioned, has a :cpp:type:`gid` to index it relative to other cells in the simulation.
+The (:cpp:type:`gid`, :cpp:type:`index`) pairs make up the :cpp:type:`cell_member` type used to index **sources**, **targets** and **gap junctions sites**
+for all cells in a simulation.
+
+
+Cell Kinds
+----------
+
+.. table:: Cell Kinds
+
+    ========================  ======================  ===========================================================
+    Identifier                Type                    Description
+    ========================  ======================  ===========================================================
+    cell kind                 enum                    * **cable**: cell with morphology described by branching
+                                                        1D cable segments.
+                                                      * **lif**: leaky-integrate and fire neuron.
+                                                      * **spiking**: spike source from values inserted via
+                                                        description.
+                                                      * **benchmark**: proxy cell used for benchmarking.
+    ========================  ======================  ===========================================================
+
+1. **Cable**
+
+   Cable cells are morphologically detailed cells represented as branching linear 1D segments. They allow the addition of
+   point and density mechanisms. They allow having *gap junctions* and *connections* to other cable cells.
```

Comment:
Does it make sense to include `:cpp:` references in the generic description? I would rather link to the C++ (later Python) specific documentation and keep it generic here.

Reply:
Good point. We have to find a way to define the "generic" type, e.g. cell_member_type, and refer to it here.

Reply:
Nora found a nice way to define a new "generic" reference type to use for definitions of types in the model documentation.